app/tests/unit/middleware/authorization.spec.js

```diff
@@ -166,7 +166,8 @@ describe('hasPermission', () => {
     ])('should call next %i times given hasDb %s and authMode %s', (nextCount, hasDb, mode) => {
       const sendCount = 1 - nextCount;
       getAppAuthModeSpy.mockReturnValue(mode);
-      getCurrentSubjectSpy.mockReturnValue(undefined);
+      //getCurrentSubjectSpy.mockReturnValue(undefined);
```

Comment:
There is still value in making sure that getCurrentSubjectSpy is tracked - although we're now using it as a nested argument call in getCurrentUserId, we should at minimum still check to make sure we're invoking getCurrentSubjectSpy at least once as an assertion. I suspect you'll want to uncomment these series of mockReturnValues, and also add in something like the following in the appropriate if block:
```js
expect(getCurrentSubjectSpy).toHaveBeenCalledTimes(someappropriatevariable);
```
In general, when we do a unit test, we want to ensure that all function calls that we mock are also asserted to be called within the function as that will tell you when the incoming code changes are no longer invoking an external function/side-effect that it used to depend on (early warning sign of potential functionality divergence).